### PR TITLE
RIFEX-45 List subdomains of a RNS domain

### PR DESCRIPTION
**Pull request by JONAF2103**
In this PR we introduce the subdomain list page, we have a list of the subdomains that belong to a domain. Also has all the code for add, delete and view subdomains, basically here we introduce all the ui to manage subdomains.

Here is a small demo:

demo-06-05-2020-jonathan-list-subdomains

This affects the services and ui layers.

### CODE REVIEW

File: ui/app/rif/components/domain-header/index.js [hunk 1]
```diff
@@ -0,0 +1,32 @@
+import React, {Component} from 'react'
+import {DomainIcon, LuminoNodeIcon, RifStorageIcon} from '../commons'
+import PropTypes from 'prop-types'
+
+class Index extends Component {
```
Comment (marcosmartinez7) [inline ui/app/rif/components/domain-header/index.js]: I think `Index ` is too ambiguous.

File: app/scripts/controllers/rif/rns/rnsjs-delegate.js [hunk 2]
```diff
@@ -125,23 +136,59 @@ export default class RnsJsDelegate extends RnsDelegate {
    * @returns {Promise<>} TransactionReceipt of the latest transaction
    */
   createSubdomain (domainName, subdomain, ownerAddress, parentOwnerAddress) {
-    return this.rnsJs.subdomains.create(domainName, subdomain, ownerAddress, parentOwnerAddress);
+    const result = this.rnsJs.subdomains.create(domainName, subdomain, ownerAddress, parentOwnerAddress);
+    result.then(transactionReceipt => {
+      if (transactionReceipt) {
```
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/rnsjs-delegate.js]: Same as https://github.com/anarancio/nifty-wallet/pull/16/files#r421147676

File: ui/app/rif/components/domain-header/index.js (hunk 1 above)
Comment (JONAF2103) [reply ui/app/rif/components/domain-header/index.js]: That was a typo, my bad didn't saw it, we can change it.

File: app/scripts/controllers/rif/rns/rnsjs-delegate.js (hunk 2 above)
Comment (JONAF2103) [reply app/scripts/controllers/rif/rns/rnsjs-delegate.js]: I reply on that PR.

File: app/scripts/controllers/rif/rns/register/index.js [hunk 3]
```diff
@@ -180,4 +182,32 @@ export default class RnsRegister extends RnsJsDelegate {
...
+   */
+  createSubdomain (domainName, subdomain, ownerAddress, parentOwnerAddress) {
+    domainName = this.addRskSuffix(domainName);
+    if (!ownerAddress) {
+      ownerAddress = this.address;
+    }
+    const node = namehash(domainName);
+    const label = web3Utils.sha3(subdomain);
+    const result = this.send(this.rnsContractInstance, 'setSubnodeOwner', [node, label, ownerAddress])
+    result.then(transactionReceipt => {
```
Comment (marcosmartinez7) [inline app/scripts/controllers/rif/rns/register/index.js]: We need to modify this in order to use transactionListener right? What is the idea? include another task for refactor?

Comment (JONAF2103) [reply app/scripts/controllers/rif/rns/register/index.js]: That code @marcosmartinez7, it's fixed on the delete PR, i'm already re-using some existent code there.

**Review by marcosmartinez7 — APPROVED** [review]
lgtm

**Pull request merged**